client/go/outline/vpn/device.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2024 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vpn
+
+import (
+	"context"
+	"errors"
+	"log/slog"
+
+	"github.com/Jigsaw-Code/outline-apps/client/go/outline/connectivity"
+	perrs "github.com/Jigsaw-Code/outline-apps/client/go/outline/platerrors"
+	"github.com/Jigsaw-Code/outline-sdk/network"
+	"github.com/Jigsaw-Code/outline-sdk/network/dnstruncate"
+	"github.com/Jigsaw-Code/outline-sdk/network/lwip2transport"
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+)
+
+// RemoteDevice is an IPDevice that connects to a remote Outline server.
+type RemoteDevice struct {
+	network.IPDevice
```

Comment:
I wonder if we should just use io.ReadWriterCloser for now and rethink if we need the IPDevice abstraction.